rfcs/docker_github_registry.md

```diff
@@ -0,0 +1,41 @@
+# RFC 198: Move the container image from Docker Hub to GitHub Registry
+
+## Summary
+
+This RFC proposes moving the container image from Docker Hub to GitHub Registry.
+
+## Details
+
+This RFC addresses the issue
+[#28903](https://github.com/web-platform-tests/wpt/issues/28903). Docker Hub
+organization allows for a limited number of members and only a few members can
+publish a new Docker image that sometimes results in the Docker image that is
+used in taskcluster being outdated as documented in [this PR](https://github.com/web-platform-tests/wpt/pull/44576#issue-2133724788).
+
+The following are the benefits of using the GitHub registry for hosting:
+
+- It is [hosted on GitHub](https://github.com/web-platform-tests/wpt/packages) together with the wpt repository.
+- Publishing is possible by any project member who has the write permissions to the
+  repository.
+- Publishing is possible [via GitHub Actions](https://github.com/web-platform-tests/wpt/blob/master/.github/workflows/docker.yml):
+
+    - The workflow can be manually invoked,
+    - or it can be further automated to deploy any changes that land.
+
+- We can use the chane to change the versioning scheme to use integer numbers.
+
+The following PR implements changes to use the image from GitHub registry
+instead of the Docker Hub one
+https://github.com/web-platform-tests/wpt/pull/46279 including modifications to
+tools/docker/frontend.py.
+
+This RFC also proposes to change the version numbers to be integers starting
+with `1`.
+
+## Risks
```

Comment:
I would guess the biggest risk is that Taskcluster pulling the image gets errors more often than with Docker Hub, or perhaps that some caching works differently so that all jobs take longer to start.

The reliability is hard to judge ahead of time, but the time it takes can be measured once at least to ensure no big difference.

Reply:
I think I can try to get the numbers from the test runs. I will add the information and update the RFC.

Reply:
I have took two random TC jobs using the old and the new image and suprisingly the download time is ~48 seconds for both (I didn't see any indication of caching in the logs):

dockerhub registry: https://community-tc.services.mozilla.com/tasks/ZNnHobGNTiS4qBiv0l7Z-A/runs/0/logs/live/public/logs/live.log

github registry: https://community-tc.services.mozilla.com/tasks/F3k9fZ6TSdC6Cd9kIwnX0A/runs/1/logs/public/logs/live.log